packages/plugins/i18n/server/src/services/localizations.ts

```diff
@@ -1,40 +1,49 @@
 import { isEmpty } from 'lodash/fp';
 
-import { async } from '@strapi/utils';
 import { getService } from '../utils';
 
-const isDialectMySQL = () => strapi.db.dialect.client === 'mysql';
-
 /**
  * Update non localized fields of all the related localizations of an entry with the entry values
- * @param {Object} entry entry to update
- * @param {Object} options
- * @param {Object} options.model corresponding model
  */
-const syncNonLocalizedAttributes = async (entry: any, { model }: any) => {
+const syncNonLocalizedAttributes = async (sourceEntry: any, model: any) => {
   const { copyNonLocalizedAttributes } = getService('content-types');
 
-  if (Array.isArray(entry?.localizations)) {
-    const nonLocalizedAttributes = copyNonLocalizedAttributes(model, entry);
-
-    if (isEmpty(nonLocalizedAttributes)) {
-      return;
-    }
-
-    const updateLocalization = (id: any) => {
-      return strapi.entityService.update(model.uid, id, { data: nonLocalizedAttributes });
-    };
-
-    // MySQL/MariaDB can cause deadlocks here if concurrency higher than 1
-    // TODO: use a transaction to avoid deadlocks
-    await async.map(
-      entry.localizations,
-      (localization: any) => updateLocalization(localization.id),
-      {
-        concurrency: isDialectMySQL() && !strapi.db.inTransaction() ? 1 : Infinity,
-      }
-    );
+  const nonLocalizedAttributes = copyNonLocalizedAttributes(model, sourceEntry);
+  if (isEmpty(nonLocalizedAttributes)) {
+    return;
+  }
+
+  const uid = model.uid;
+  const documentId = sourceEntry.documentId;
+  const status = sourceEntry?.publishedAt ? 'published' : 'draft';
+
+  // Find all the entries that need to be updated
+  // Every other entry of the document in the same status but a different locale
+  const entriesToUpdate = await strapi.db.query(uid).findMany({
+    where: {
+      documentId,
+      publishedAt: status === 'published' ? { $ne: null } : null,
+      locale: { $ne: sourceEntry.locale },
+    },
+    select: ['locale', 'id'],
+  });
+
```

Comment:
Planning for the next ask: support non localized relations. It will mean copying relations (using the documentId probably) @Marc-Roig can help on this aspect

Reply:
next ask of the next ask. We might want tnot to support non localized relations that are nested in compos